src/citrine/resources/material_run.py

```diff
@@ -163,10 +126,20 @@ def get_history(self, scope, id) -> Type[MaterialRun]:
         base_path = os.path.dirname(self._get_path(ignore_dataset=True))
         path = base_path + "/material-history/{}/{}".format(scope, id)
         data = self.session.get_resource(path)
-        # Rehydrate a taurus object based on the data
-        model = loads(json.dumps([data['context'], data['root']], cls=TaurusEncoder))
-        # Convert taurus objects into citrine-python objects
-        return MaterialRun.build(model)
+
+        # Add the root to the context and sort by writable order
+        blob = dict()
+        blob["context"] = sorted(
+            data['context'] + [data['root']],
+            key=lambda x: writable_sort_order(x["type"])
+        )
+        root_uid = next(iter(data['root']['uids'].items()))
+        # Add a link to the root as the "object"
+        blob["object"] = LinkByUID(scope=root_uid[0], id=root_uid[1])
+
+        # Serialize using normal json (with the TaurusEncoder) and then deserialize with the
+        # TaurusJson encoder in order to rebuild the material history
+        return MaterialRun.get_encoder().loads(json.dumps(blob, cls=TaurusEncoder, sort_keys=True))
```

Comment:
Is there a way to move from python primitive types to citrine-python types directly? In my dream world `git grep 'loads.*dumps'` exits with failure.

Reply:
I think we'd want to dive into `get_resource` to deserialize using `get_json_support` the first time rather than going round-robin through the string again.  Let's leave that out of scope for this PR, though?  It'd be an easy non-breaking change to make in the future.